lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -199,13 +200,23 @@ impl TableOfContent {
             drop(removed);
 
             // Move collection to ".deleted" folder to prevent accidental reuse
-            let uuid = Uuid::new_v4().to_string();
+            // the original collection path will be moved atomically within this
+            // directory.
             let removed_collections_path =
                 Path::new(&self.storage_config.storage_path).join(".deleted");
             tokio::fs::create_dir_all(&removed_collections_path).await?;
-            let deleted_path = removed_collections_path
-                .join(collection_name)
-                .with_extension(uuid);
+
+            // Generate a random alphanumeric string to append to the deleted storage directory.
+            let random_str: String = rand::thread_rng()
+                .sample_iter(&Alphanumeric)
+                // Limit the generated string to 8 characters to avoid issues
+                // with file name length restrictions. This size provides a
+                // very low probability of collision (1 in 62^8).
+                .take(8)
+                .map(char::from)
+                .collect();
+            let deleted_path = removed_collections_path.join(random_str);
```

Comment:
Though the chance for collision is low, I'd still prefer to totally eliminate it from happening.

Maybe we can use [`TempDir`](https://docs.rs/tempfile/latest/tempfile/) here instead, we already use that crate across the codebase.

It'll generate a random directory name for us and ensures uniqueness. As a bonus it'll also help us clean up the directory if we some how failed to do so in the tokio spawn loop below.

I'd suggest to use the same `.deleted` directory like now so you might have to use [`tempdir_in`](https://docs.rs/tempfile/latest/tempfile/fn.tempdir_in.html).

What do you think?

Reply:
makes sense to me. both reasoning are strong; I will send a new revision to it. 

fyi: ```tempdir``` will produce random file names of 12 chars, which extend the collision factor to 1 in 62^12 (that's 3.23 quintillion)

Reply:
> tempdir will produce random file names of 12 chars, which extend the collision factor to 1 in 62^12 (that's 3.23 quintillion)

tempdir also checks for collision internally, pick a new random name if it already exists. So no concern there.

If we can reduce the random characters to 8 for tempdir that's fine by me. No strong opinions there.